.ci/openshift-ci/common.sh

```diff
@@ -0,0 +1,238 @@
+    -H 'Content-Type: application/json' \
+    -d '{
+    "url": "'${GIT_REPO_URL}'"
+  }'
+}
+
+# check that factory resolver returns correct value
+testFactoryResolverNoPatOAuth() {
+  PUBLIC_REPO_URL=$1
+  PRIVATE_REPO_URL=$2
+
+  echo "[INFO] Check factory resolver with NO PAT/OAuth setup"
+  # for public repo
+  if [ "$(requestFactoryResolverGitRepoUrl ${PUBLIC_REPO_URL} | grep "HTTP/1.1 200")" ]; then
+    echo "[INFO] Factory resolver returned 'HTTP/1.1 200' status code."
+  else
+    echo "[ERROR] Factory resolver returned wrong status code. Expected: HTTP/1.1 200."
+    exit 1
+  fi
+
+  # for private repo
+  if [ "$(requestFactoryResolverGitRepoUrl ${PRIVATE_REPO_URL} | grep "HTTP/1.1 400")" ]; then
+    echo "[INFO] Factory resolver returned 'HTTP/1.1 400' status code. Expected client side error."
+  else
+    echo "[ERROR] Factory resolver returned wrong status code. Expected 'HTTP/1.1 400'."
+    exit 1
+  fi
+}
+
+testFactoryResolverWithPatOAuth() {
+  PUBLIC_REPO_URL=$1
+  PRIVATE_REPO_URL=$2
+
+  echo "[INFO] Check factory resolver with PAT/OAuth setup"
+  # for public repo
+  if [ "$(requestFactoryResolverGitRepoUrl ${PUBLIC_REPO_URL} | grep "HTTP/1.1 200")" ]; then
+    echo "[INFO] Factory resolver returned 'HTTP/1.1 200' status code."
+  else
+    echo "[ERROR] Factory resolver returned wrong status code. Expected: HTTP/1.1 200"
+    exit 1
+  fi
+
+  # for private repo
+  if [ "$(requestFactoryResolverGitRepoUrl ${PRIVATE_REPO_URL} | grep "HTTP/1.1 200")" ]; then
+    echo "[INFO] Factory resolver returned 'HTTP/1.1 200' status code."
+  else
+    echo "[ERROR] Factory resolver returned wrong status code. Expected: HTTP/1.1 200"
+    exit 1
+  fi
+}
+
+runTestWorkspaceWithGitRepoUrl() {
+  WS_NAME=$1
+  PROJECT_NAME=$2
+  GIT_REPO_URL=$3
+
+  oc new-project ${USER_PROJECT_NAME} || true
+  oc project ${USER_PROJECT_NAME}
+  cat .ci/openshift-ci/devworkspace-test.yaml > devworkspace-test.yaml
+
+  # patch the devworkspace-test.yaml file
+  sed -i "s#ws-name#${WS_NAME}#g" devworkspace-test.yaml
+  sed -i "s#project-name#${PROJECT_NAME}#g" devworkspace-test.yaml
+  sed -i "s#git-repo-url#${GIT_REPO_URL}#g" devworkspace-test.yaml
+
+  cat devworkspace-test.yaml
+
+  oc apply -f devworkspace-test.yaml -n ${USER_PROJECT_NAME}
+  oc wait -n ${USER_PROJECT_NAME} --for=condition=Ready dw ${WS_NAME} --timeout=360s
+}
+
+testProjectIsCloned() {
+  PROJECT_NAME=$1
+  WORKSPACE_POD_NAME=$(oc get pods -n ${USER_PROJECT_NAME} | grep workspace | awk '{print $1}')
+  if oc exec -it -n ${USER_PROJECT_NAME} ${WORKSPACE_POD_NAME} -- test -f /projects/${PROJECT_NAME}/${YAML_FILE_NAME}; then
+    echo "[INFO] Project file /projects/${PROJECT_NAME}/${YAML_FILE_NAME} exists."
+  else
+    echo "[ERROR] Project file /projects/${PROJECT_NAME}/${YAML_FILE_NAME} does not exist."
+    exit 1
+  fi
+}
+
+# check a project is not cloned for private repo with no OAuth/PAT setup
+testProjectIsNotCloned() {
```

Comment:
This method seems to be duplicating testProjectIsCloned() method.

Reply:
Done.